diff --git a/makefile b/makefile
index fe829dc..9043ab1 100644
--- a/makefile
+++ b/makefile
@@ -24,15 +24,14 @@ build/boot/load.bin: build src/boot/load.asm src/link/load.ld
 	$(LD) -T src/link/load.ld -o build/boot/load.elf build/boot/load.o
 	$(OC) build/boot/load.elf build/boot/load.bin
 
-# Compile the bootloader, second stage
-build/boot/start.bin: build src/boot/start.asm src/link/start.ld build/boot/load.bin
-	$(AS) src/boot/start.asm -o build/boot/start.o
-	$(LD) -T src/link/start.ld -o build/boot/start.elf build/boot/start.o
-	$(OC) build/boot/start.elf build/boot/start.bin
+build/kernel/kmalloc.o: build src/kernel/kmalloc.asm
+	 $(AS) src/kernel/kmalloc.asm -o build/kernel/kmalloc.o
+
+build/kernel/entry.o: build src/kernel/entry.c
+	 $(CC) -c src/kernel/entry.c -o build/kernel/entry.o
 
 # Compile the kernel to a flat binary
-build/kernel/kernel.bin: build src/link/kernel.ld src/kernel/entry.c
-	$(CC) -c src/kernel/entry.c -o build/kernel/entry.o
+build/kernel/kernel.bin: build src/link/kernel.ld build/kernel/entry.o build/kernel/kmalloc.o
 	$(LD) -T src/link/kernel.ld -o build/kernel/kernel.bin
 
 # Create the floppy disc system image
diff --git a/src/kernel/entry.c b/src/kernel/entry.c
index a4e69fc..7977718 100644
--- a/src/kernel/entry.c
+++ b/src/kernel/entry.c
@@ -1,8 +1,11 @@
 
 #include "types.h"
 
+
 volatile bool truth = true;
 
+//extern char test();
+
 void start() {
 	uint16_t volatile* screen = (uint16_t*) 0xB8000;
 
diff --git a/src/link/kernel.ld b/src/link/kernel.ld
index 6c6427a..8c70fb5 100644
--- a/src/link/kernel.ld
+++ b/src/link/kernel.ld
@@ -3,6 +3,7 @@ ENTRY(start)
 
 /* Files that need to be linked together */
 INPUT(build/kernel/entry.o)
+INPUT(build/kernel/kmalloc.o)
 
 SECTIONS {
 	. = 0x8000;
